dist/jquery.mark.es6.js

```diff
@@ -502,90 +510,42 @@ class Mark {
   isNumeric(value) {
     return Number(parseFloat(value)) == value;
   }
-  checkRanges(array) {
-    if (
-      !Array.isArray(array) ||
-      Object.prototype.toString.call(array[0]) !== '[object Object]'
-    ) {
-      this.log('markRanges() will only accept an array of objects');
-      this.opt.noMatch(array);
-      return [];
-    }
-    const stack = [];
-    let last = 0;
-    array
-      .sort((a, b) => {
-        return a.start - b.start;
-      })
-      .forEach(item => {
-        let {start, end, valid} = this.callNoMatchOnInvalidRanges(item, last);
-        if (valid) {
-          item.start = start;
-          item.length = end - start;
-          stack.push(item);
-          if ( !this.opt.wrapAllRanges) {
-            last = end;
-          }
+  isObject(obj) {
+    return Object.prototype.toString.call(obj) === '[object Object]';
+  }
+  isArrayOfObjects(array) {
+    return Array.isArray(array) && array.some(item => this.isObject(item));
+  }
+  checkRanges(array, logs, max) {
+    const level = 'error';
+    const ranges = array.filter(range => {
+      let valid = false;
+      if (this.isNumeric(range.start) && this.isNumeric(range.length)) {
```

Comment:
I'm a fan of the library work you're doing and I use the library right after doing something very similar to this check. I think you can get some more speed out of using the built-in method `Number.isInteger()` instead of continuing to use the `isNumeric()` function above and the `parseInt()` built-in below. 

Using the `isNumeric()` function as written above means that non-integer values, like `15.3`, will return `true` whereas `Number.isInteger()` would return `false` for `15.3`. My understanding of DOM ranges is that they are always a pair of boundary points, which themselves are a tuple of a node and an unsigned integer. Assuming the library doesn't coerce an incoming range argument to string during processing, I don't think it should ever receive a valid range containing a float. 

If that's true, then lines 523, 524, and 525 could be simplified down to a single line: ```if (Number.isInteger(range.start) && Number.isInteger(range.length)) {``` 

Does this make sense to you? Sorry if I'm missing a step.

Reply:
When I rewrite the code that works many years without any problem in order to simplify, my main care is not to break the code logic and keep the backward compatibility as much as possible.
Besides, to wrap ~28000 ranges in Firefox take ~320 ms, your suggestion gives ~2.5 ms gain, but break the backward compatibility e.g. the ranges `{ start: 10, length: 15.3 }, { start: '50', length: 10 }` won't be highlighted.  
Ranges are often written manually - mistakes are possible.

In this library I break backward compatibility in filter callbacks (the parameters that count mark elements change to count matches - can break code that uses `acrossElements` option) but there is a strong reason that the new code gain more from those changes (the old code rare if any uses these parameters).

Reply:
Ah, ok, I was thinking we'd want to throw if an invalid range was passed in. If the goal is to _avoid_ throwing even when a string or float is passed in, this makes perfect sense and you're 100% right. Thank you for explaining and for your work on the library!

My use case is near real-time so I'm always trying to shave a few ms off the top when I can. If we want to avoid throwing for strings and floats while staying inside the loop as much as we can, would the unary plus and `Math.trunc()` work while still keeping backward compatibility? Like this: `if (Number.isInteger(Math.trunc(+range.start)) && Number.isInteger(Math.trunc(+range.length))) {`

Reply:
And another thing I forgot to mention - `Number.isInteger()` breaks IE 11 support; `Math.trunc()` will breaks some other older browsers.
In `getPatterns()` method, I'm also using ` isNumeric & parseInt` in order not to break IE support.

From a practical point of view, to highlight a dozen or two ranges (probably the most common case) this is nano if not pico optimization considering the browser loading, parsing time, plus wrapping time, plus browser rendering time.
It gives more headache than performance gain.

By the way, the unary plus is internally implemented in `Math.trunc()`. And, as it returns the integer or NaN it's probably shorter to use `isNaN()` than `Number.isInteger()`.

Reply:
Although in polyfill the unary plus is implemented, when value is a string, without plus it significantly slowly.
``` js
Math.trunc = function(v) {
  v = +v;
  return (v - v % 1) || (!isFinite(v) || v === 0 ? v : v < 0 ? -0 : 0);
};
```
And another problem - when value is `null` it returns `0`.  
`isNaN()` isn't affects performance.

Reply:
Sounds like you've got a really good grip on all of this so I'll stop tugging on your shirt tail like a little brother :) 

I'm obviously not as skilled as you but I am improving quickly, so if you find something where you'd like a bit of help, or maybe someone to bounce ideas off, just post a note in the project and I'll do what I can.

Thanks again for your efforts!